### PR TITLE
Removed listings state from searchresultscreen, connected with redux

### PR DESCRIPTION
**Pull request by nejohnston**

### CODE REVIEW

File: js/screens/Listing/ListingScreen.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import React, { Component } from 'react';
```
Comment (siddharthparmar7) [inline js/screens/Listing/ListingScreen.js:1]: Please put JSX from here to Listing.js and use Listing.js from this file.

File: js/screens/Listing/ListingScreen.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+import PropTypes from 'prop-types';
+
+import { View, Text, Button } from 'react-native';
+
+class ListingScreen extends Component {
+  static propTypes = {
```
Comment (siddharthparmar7) [inline js/screens/Listing/ListingScreen.js]: Use either ES6 or ES7 to add propTypes.

File: js/screens/Listing/ListingScreen.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    navigation: PropTypes.object
```
Comment (siddharthparmar7) [inline js/screens/Listing/ListingScreen.js]: Please use .isRequired with navigation props.

File: js/screens/LocationSearch/LocationSearchScreen.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import React, { Component } from 'react';
+import LocationSearch from './LocationSearch';
+import PropTypes from 'prop-types';
+
+class LocationSearchScreen extends Component {
+  constructor() {
+    super();
+    this.state = {
```
Comment (siddharthparmar7) [inline js/screens/LocationSearch/LocationSearchScreen.js]: Please use Redux not component state.

File: js/screens/Login/Login.js [hunk 5]
```diff
@@ -45,7 +68,8 @@ const Login = ({ handleEmail, handlePassword, handleSubmit }) => {
 Login.propTypes = {
   handleSubmit: PropTypes.func.isRequired,
   handleEmail: PropTypes.func.isRequired,
-  handlePassword: PropTypes.func.isRequired
+  handlePassword: PropTypes.func.isRequired,
+  navigation: PropTypes.object
```
Comment (siddharthparmar7) [inline js/screens/Login/Login.js:72]: please use .isRequired with this propType.

File: js/screens/Login/LoginScreen.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+import { firebaseAuth } from '../../config/firebaseConfig';
+
+import { AsyncStorage } from 'react-native';
+import Login from './Login';
+import PropTypes from 'prop-types';
+
+class LoginScreen extends Component {
+  constructor() {
+    super();
+    this.state = {
```
Comment (siddharthparmar7) [inline js/screens/Login/LoginScreen.js:14]: Please use the Redux store.

File: js/screens/Login/LoginScreen.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+      emailInput: '',
+      passwordInput: '',
+      error: ''
+    };
+  }
+  static navigationOptions = {
+    header: null
+  };
+  static propTypes = {
+    navigation: PropTypes.object
```
Comment (siddharthparmar7) [inline js/screens/Login/LoginScreen.js]: Please use .isRequired with navigation prop.

File: js/screens/Profile/Profile.js [hunk 8]
```diff
@@ -1,19 +1,29 @@
...
+        style={styles.profileText}
+      />
+      <Button onPress={signOut} title="sign me out" />
     </View>
   );
 };
 
+Profile.propTypes = {
+  navigation: PropTypes.object,
+  signOut: PropTypes.func
```
Comment (siddharthparmar7) [inline js/screens/Profile/Profile.js:26]: Please use .isRequired, if not then add default props for it.

File: js/screens/Profile/ProfileScreen.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+import { Button, AsyncStorage } from 'react-native';
+import PropTypes from 'prop-types';
+
+import { colors } from '../../config/styles';
+
+import Profile from './Profile';
+
+class ProfileScreen extends Component {
+  static propTypes = {
+    navigation: PropTypes.object
```
Comment (siddharthparmar7) [inline js/screens/Profile/ProfileScreen.js]: Please use .isRequired with navigation prop

File: js/screens/SearchResult/SearchResult.js [hunk 10]
```diff
@@ -29,7 +39,8 @@ const SearchResult = ({ listings }) => {
 };
 
 SearchResult.propTypes = {
-  listings: PropTypes.array.isRequired
+  listings: PropTypes.array.isRequired,
+  navigation: PropTypes.object
```
Comment (siddharthparmar7) [inline js/screens/SearchResult/SearchResult.js:43]: please add .isRequired to navigation prop

File: js/screens/SignUpName/SignUpName.js [hunk 11]
```diff
@@ -32,7 +50,8 @@ const SignUpName = ({ handleFirstName, handleLastName }) => {
 
 SignUpName.propTypes = {
   handleFirstName: PropTypes.func.isRequired,
-  handleLastName: PropTypes.func.isRequired
+  handleLastName: PropTypes.func.isRequired,
+  navigation: PropTypes.object
```
Comment (siddharthparmar7) [inline js/screens/SignUpName/SignUpName.js:54]: Please add .isRequired to nav prop

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Thank you!

File: js/screens/Listing/ListingScreen.js (hunk 1 above)
Comment (nejohnston) [reply js/screens/Listing/ListingScreen.js:1]: done

File: js/screens/Listing/ListingScreen.js (hunk 2 above)
Comment (nejohnston) [reply js/screens/Listing/ListingScreen.js]: removed all es7 proptypes from file

File: js/screens/Listing/ListingScreen.js (hunk 3 above)
Comment (nejohnston) [reply js/screens/Listing/ListingScreen.js]: added isRequired to navigation props

File: js/screens/Login/LoginScreen.js (hunk 6 above)
Comment (nejohnston) [reply js/screens/Login/LoginScreen.js:14]: this is being handled by colin

**Pull request closed without merging**